### PR TITLE
[API-14303] - ClaimsApi - Fix 526 Supporting Doc Uploads to EVSS

### PR DESCRIPTION
**Pull request by jw81**
## Description of change

- fixes an error in API-14303 where we're sending blank `auth_headers` to EVSS during document uploads
- fixes an error in API-14302 where a Claim's `file_data` is removed before we've had a chance to upload the PDF to EVSS
- removes logic that deletes uploads from S3 until we feel certain that EVSS doc uploads are stable

## Original issue(s)
[API-14303](https://vajira.max.gov/browse/API-14303) and [API-14302](https://vajira.max.gov/browse/API-14302)

### CODE REVIEW

File: modules/claims_api/spec/workers/claim_establisher_spec.rb [hunk 1]
```diff
@@ -36,9 +36,6 @@
     subject.new.perform(claim.id)
     claim.reload
     expect(claim.evss_id).to eq(1337)
-    expect(claim.form_data).to be_empty
-    expect(claim.auth_headers).to be_empty
-    expect(claim.file_data).to be_nil
```
Comment (jw81) [inline modules/claims_api/spec/workers/claim_establisher_spec.rb:41]: I moved these checks to a better set of specs on the model (see `auto_established_claim_spec.rb`)

File: modules/claims_api/app/models/claims_api/auto_established_claim.rb [hunk 2]
```diff
@@ -235,12 +235,12 @@ def validate_service_dates
       end
     end
 
+    # do not clear out 'auth_headers' or 'file_data' attributes
+    # 'auth_headers' is required to make calls to EVSS for uploading docs
+    # 'file_data' is required to know what doc to upload to EVSS
+    # See API-14303
     def remove_encrypted_fields
```
Comment (hinzed1127) [inline modules/claims_api/app/models/claims_api/auto_established_claim.rb:242]: just to add more explicitness to the `after_validation` statement [this is called from](https://github.com/department-of-veterans-affairs/vets-api/pull/9466/files#diff-cecc476b1562949c65821a873933ee4c2cf286b5023f0e6b06f97c2b3fc5a3cbL22)

File: modules/claims_api/spec/workers/claim_establisher_spec.rb (hunk 1 above)
Comment (hinzed1127) [reply modules/claims_api/spec/workers/claim_establisher_spec.rb:41]: nice, good additions

**Review by hinzed1127 — APPROVED** [review]

File: modules/claims_api/app/models/claims_api/auto_established_claim.rb (hunk 2 above)
Comment (jw81) [reply modules/claims_api/app/models/claims_api/auto_established_claim.rb:242]: I really do like that idea.  I think in the interest of getting a fix out there ASAP, we should maybe handle that in another upcoming ticket/PR.  If that's cool?

Comment (hinzed1127) [reply modules/claims_api/app/models/claims_api/auto_established_claim.rb:242]: Sounds good to me!

**Pull request merged**